### DISCUSSION TITLE
FlowControl clarification

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#301] [Help & Questions] [answered]
I am 90% sure here, but I've been documenting the C# code and need to be certain. 

My understanding is FlowControl is designed to manually control the RNode's broadcasts. If it's enabled the radio's Interface_Ready is set to false, preventing further transmissions until reset when Process_Queue is called. But since Process_Queue sets it true, the only case where a Process_Outgoing isn't called by a Process_Queue, hence the only time it could possibly be false, is at the end of the ReceiveLoop where the ID_Callsign is boradcast.

Did I make a mistake, or is, at this time, the only use for FlowControl to prevent the ID_Callsign from being broadcast? I am simply asking from an RNode standpoint, I understand this may have implications elsewhere in the Reticulum ecosystem or with other interfaces.

**Comment by markqvist** [comment, marked as answer]
It's a bit of an artifact from when some people made RNodes using microcontrollers and dev boards with very little RAM, like 2KB. In those cases, there was not really enough RAM to keep a sufficient amount of packets in an onboard packet buffer, so the host system would have to take care of that instead.

When the packet buffer on the RNode was full, the host system would buffer packets until the RNode was ready to receive more over USB/serial.

This is likely going away in a cleanup in the near future, so you can safely ignore it. The future firmwares will only support boards that have enough RAM for decent packet buffers.

**Reply by faragher** [reply]
Thanks. I'll just leave it as-is until it's officially removed. It doesn't seem to be hurting anything.
